woocommerce/woocommerce-gateway-stripe
# Update the disconnect account message

Fixes #2943

## Changes proposed in this Pull Request:

This PR updates the Disconnect Modal text to clarify that after disconnecting the account, when completing the onboarding a new Stripe account might be required.

It also adds tracks for the `disconnect` and `manual key update` events.

## Testing instructions
0. Make sure you have opted into tracking under the `WC settings > Advanced > WooCommerce.com`
    - /wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com
    - The events may take ~10 minutes to show up in Tracks.
1. Go to the Stripe plugin settings page:
    - /wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings
2. In the `Account Settings` section, click the 3 dots, and the click `Edit account keys`
    ![Screenshot 2024-02-26 at 16 54 10](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/c5caefff-9829-46d5-90a1-9665773961b5)
3. Enter the keys from your stripe account dashboard and save
4. In Tracks, confirm that an event called `wcstripe_stripe_keys_updated` was recorded
5. Again, in the `Account Settings` section click the 3 dots, and this time click the `Disconnect` option
6. Check that the text in the modal includes: "... Due to this change, you might be required to create a new Stripe account should you want to reactivate."
    ![Screenshot 2024-02-26 at 17 01 32](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/a7772cf7-9d05-4719-a16c-2a9926dfaa2c)
7. Click on the red "Disconnect" button
8. After the pages reloads, it redirects to the onboarding:
    ![Screenshot 2024-02-26 at 17 02 35](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/fc5db3d3-17b9-400a-8e9a-5b360e376947)
9. In Tracks, confirm that an event called `wcstripe_stripe_disconnected` was recorded
